docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,421 @@
+6. Factory merges are applied (see `Optimizers`_ for explanation).
+
+This results in a *regular* state dict with plain tensors that can be further processed by the application (which usually means running ``model.load_state_dict(state_dict)``.
+
+
+dist_checkpointing.load_common_state_dict
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_common_state_dict`` function is an entry point that allows loading only the “common” part of the checkpoints.
+Most of the checkpoint config and metadata can be loaded with this method, which allows skipping data loading in order to take decisions regarding checkpoint config, version, etc.
+
+dist_checkpointing.load_tensors_metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_tensors_metadata`` function is an entry point that allows reading all ShardedTensors metadata from the checkpoint without loading any data.
+The result is a sharded state dict with trivial sharding (every tensor is sharded into one big shard).
+
+dist_checkpointing.load_plain_tensors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_plain_tensors`` function is an entry point that allows reading sharded tensors stored in the checkpoint without any sharding (as plain tensors).
+This function is simply a composition of ``load_tensors_metadata`` and ``save``.
+
+Save and Load Strategies
+------------------------
+There are multiple ways to save a sharded state dict into a serialized checkpoint. They can be provided by the user as saving and loading strategies.
+
+There are four types of strategies:
+
+1. Saving strategy for ShardedTensors
+2. Saving strategy for "common" data
+3. Loading strategy for ShardedTensors
+4. Loading strategy for "common" data
+
+Additionally, ShardedObjects are handled with either "sharded" or "common" strategy depending on its capabilities (``can_handle_sharded_objects`` property).
+
+Each saving strategy is associated with a ``backend`` and a ``version``.
+Each loading strategy can be associated with multiple ``backend``s and ``version``s it can load.
+For a given backend and version, the composition of every saving and loading strategy **must be functionally equivalent**.
+Strategies are the main way to introduce optimizations to the saving and loading algorithm without altering the checkpoint format.
+
+In the following example, the "fully parallel" wrappers modify the saving and loading *algorithm*, but the underlying checkpoint *format* (and ``backend`` in consequence) stays the same.
+It makes the ``basic_save_load`` and ``fully_parallel_save_load`` functions equivalent:
+
+.. code-block:: python
+
+    from megatron.core import dist_checkpointing
+    from megatron.core.dist_checkpointing.strategies.torch import TorchDistLoadShardedStrategy, TorchDistSaveShardedStrategy
+    from megatron.core.dist_checkpointing.strategies.fully_parallel import FullyParallelLoadStrategyWrapper, FullyParallelSaveStrategyWrapper
```

Comment:
maybe use parentheses here to make the lines shorter. otherwise they go out of view. similarly some of the lines in `full_parallel_save_load` are too long